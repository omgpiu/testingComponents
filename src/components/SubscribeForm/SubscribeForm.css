.container {
    display: flex;
    flex-direction: column;
    align-self: center
}

.form {
    display: flex;
    flex-direction: column;
    align-self: center;
    width: 520px
}

.form button {
    width: 300px;
}
