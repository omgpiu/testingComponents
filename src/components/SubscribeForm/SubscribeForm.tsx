import React, { useState } from 'react';
import './SubscribeForm.css';

const SubscribeForm = () => {

    const [isDisabled, setIsDisabled] = useState(true);
    const [email, setEmail] = useState('');

    function handleChange(e: any) {
        setEmail(e.target.value);
        setIsDisabled(e.target.value === '');
    }

    return (
        <div className="container">
            <h1>Subscribe  Newsletter</h1>
            <form className="form">
                <label htmlFor="email">Email Address</label>
                <input onChange={ handleChange }
                       data-testid="email"
                       type="email" id="email" name="email" placeholder="Email Address"
                       value={ email }/>
                <input type="checkbox" name="agreement_checkbox" id="agreement_checkbox"/>
                <label htmlFor="agreement_checkbox">I agree to disagree whatever the terms and conditions are.</label>
                <button name="subscribe-button" type="submit" className="button" disabled={ isDisabled }>Subscribe
                </button>
            </form>
        </div>
    );
};

export default SubscribeForm;
