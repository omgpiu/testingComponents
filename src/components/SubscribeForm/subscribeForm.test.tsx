import { render, screen } from '@testing-library/react';
import SubscribeForm from './SubscribeForm';
import userEvent from '@testing-library/user-event';
import TestRenderer from 'react-test-renderer';


it('The subscribe button is disabled before typing anything in the input text box', () => {
    render(<SubscribeForm/>);

    expect(screen.getByRole('button', { name: /subscribe/i })).toBeDisabled();
    expect(screen.getByRole('checkbox', { name: 'I agree to disagree whatever the terms and conditions are.' })).toBeInTheDocument()
    expect(screen.getByRole('textbox', { name: /email/i })).toBeInTheDocument()
    expect(screen.getByTestId('email')).toBeInTheDocument()
});
it('The subscribe button becomes enabled when we start typing in the input text box', () => {
    render(<SubscribeForm/>);

    userEvent.type(screen.getByRole('textbox', { name: /email/i }), 'abc@email.com');
    expect(screen.getByRole('button', { name: /subscribe/i })).toBeEnabled();
});
it("Test to match snapshot of component", () => {
    const subscribeFormTree = TestRenderer.create(<SubscribeForm />).toJSON();
    expect(subscribeFormTree).toMatchSnapshot();
})


export default 1
