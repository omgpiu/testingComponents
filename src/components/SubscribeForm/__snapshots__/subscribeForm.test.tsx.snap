// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to match snapshot of component 1`] = `
<div
  className="container"
>
  <h1>
    Subscribe  Newsletter
  </h1>
  <form
    className="form"
  >
    <label
      htmlFor="email"
    >
      Email Address
    </label>
    <input
      data-testid="email"
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="email"
      value=""
    />
    <input
      id="agreement_checkbox"
      name="agreement_checkbox"
      type="checkbox"
    />
    <label
      htmlFor="agreement_checkbox"
    >
      I agree to disagree whatever the terms and conditions are.
    </label>
    <button
      className="button"
      disabled={true}
      name="subscribe-button"
      type="submit"
    >
      Subscribe
    </button>
  </form>
</div>
`;
